src/syntax/abs/trans.rs

```diff
@@ -67,35 +66,30 @@ fn trans_expr_inner(
                 Err(TCE::LookUpFailed(ident.clone()))
             }
         }
-        Expr::App(app_vec) => app_vec
-            .iter()
-            .try_fold(None::<Abs>, |result, each_expr| -> TCM<_> {
+        Expr::App(applied, app_vec) => app_vec.iter().try_fold(
```

Comment:
So I've stored the first element inside of `Expr::App` (and so does the others) to guarantee that it's indeed not empty from the type-system's perspective.